Newsletters: add 361 (2025-07-04) (#2402)
bitcoinops/bitcoinops.github.io: _posts/en/newsletters/2025-07-04-newsletter.md

  different R value for the signature.  These issues are independent of
  the update complexity and peer protocol updates Gregory Sanders
  describes above."

- **Vault output script descriptor:** Sjors Provoost [posted][] to
  Delving Bitcoin to discuss how the recovery information for a wallet
  using [vaults][] could be specified using an [output script
  descriptor][].  In particular, he focused on vaults based on
  [OP_CHECKTEMPLATEVERIFY][] (CTV), such as those provided by James
  O'Beirne's [simple-ctv-vault][] proof-of-concept implementation (see
  [Newsletter #191][news191 simple-ctv-vault]).

  Provoost cited a [comment][] from a previous discussion by Salvatore
  Ingala that said, "my general take is that descriptors are the wrong
  tool for this purpose"---a sentiment Sanket Kanjalkar [agreed with][]
  in the current thread but [found][] a potential workaround for.
  Kanjalkar described a variant CTV-based vault where funds are
  deposited into a more typical descriptor and, from there, are moved
  into a CTV vault.  This avoids a situation that could lead naive users
  into losing money and also allows the creation of a descriptor that

>   assumes all funds to the typical descriptor are moved into a vault the

Reviewer: ```suggestion
  assumes all funds in the typical descriptor are moved into a vault with the
```

    legacy script support could save about 8 vbytes in some cases, but
    Sanders [linked][] to his previous pay-to-CTV (P2CTV) proposal and
    proof-of-concept implementation that makes this savings available in
    witness script.

  - *Limits of CTV-only vault support:* signatory Jameson Lopp [noted][]
    that he's "most interested in vaults," starting a discussion about
    the set of properties CTV vaults would provide, how they compare to
    vaults deployable today using presigned transactions, and whether
    they provide a meaningful improvement in security (especially
    compared to more advanced vaults that would require additional
    consensus changes).  Key takeaways from this discussion included:

    - *Address reuse danger:* both presigned and CTV vaults must prevent
      users from reusing vaulting addresses or funds are likely to be
      lost.  One way this can be accomplished is by a two-step vaulting
      procedure requiring two onchain transactions to deposit funds into
      the vault.  More advanced vaults requiring additional consensus
      changes would not have this problems, allowing deposit even to a

>       reused address (althought that would, of course, [reduce

Reviewer: ```suggestion
      reused address (although that would, of course, [reduce
```

  [posted][osuntokun onion] to Delving Bitcoin about allowing nodes to
  use separate connections for relaying [onion messages][topic onion messages] than they
  use for relaying [HTLCs][topic htlc].  Although separate connections may
  currently be possible, such as in the case of direct relay (see
  Newsletters [#283][news283 oniondirect] and [#304][news304
  onionreply]), Osuntokun suggests that separate connections
  should always be an option, allowing nodes to have a different set of
  peers for onion messages from the set of peers they use for relaying
  payments.  He makes several arguments in support of this alternative
  approach: it more clearly separates concerns, nodes can cheaply
  support a greater density of onion message peers than channel peers
  (as channels cost money to create), separation may allow deployment of
  privacy-improving key rotation, and separation may allow faster
  delivery of onion messages due to not being blocked by the
  HTLC commitment communication protocol.  Osuntokun provides specific
  details about the proposed protocol.

  A concern of several replying developers was how the network for onion

>   message would prevent
>   nodes from being flooded by excessive number of peers.  In current

Reviewer 2: ```suggestion
  messages could allow
  nodes to be flooded by an excessive number of peers.  In current
```

@harding
